core/src/main/java/org/testcontainers/utility/DockerLoggerFactory.java

```diff
@@ -1,26 +1,50 @@
 package org.testcontainers.utility;
 
+import java.lang.reflect.Proxy;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
 
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class DockerLoggerFactory {
+    private static final ConcurrentMap<String, Logger> cache = new ConcurrentHashMap<>();
 
     public static Logger getLogger(String dockerImageName) {
+        return getLogger("org.testcontainers", dockerImageName);
+    }
+
+    public static Logger getLogger(Class<?> source, String dockerImageName) {
+        return getLogger(source.getName(), dockerImageName);
+    }
+
+    public static Logger getLogger(String source, String dockerImageName) {
+        return cache.computeIfAbsent(source + "/" + dockerImageName, name -> {
+            final String abbreviatedName;
+            if (dockerImageName.contains("@sha256")) {
+                abbreviatedName = dockerImageName.substring(0, dockerImageName.indexOf("@sha256") + 14) + "...";
+            } else {
+                abbreviatedName = dockerImageName;
+            }
+
+            final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+            final Logger logger = LoggerFactory.getLogger(source);
 
-        final String abbreviatedName;
-        if (dockerImageName.contains("@sha256")) {
-            abbreviatedName = dockerImageName.substring(0, dockerImageName.indexOf("@sha256") + 14) + "...";
-        } else {
-            abbreviatedName = dockerImageName;
-        }
-
-        if ("UTF-8".equals(System.getProperty("file.encoding"))) {
-            return LoggerFactory.getLogger("\uD83D\uDC33 [" + abbreviatedName + "]");
-        } else {
-            return LoggerFactory.getLogger("docker[" + abbreviatedName + "]");
-        }
+            return (Logger) Proxy.newProxyInstance(
+                classLoader,
+                new Class[]{
+                    Logger.class
+                },
+                (proxy, method, args) -> {
+                    try (MDC.MDCCloseable c = MDC.putCloseable("image.name", abbreviatedName)) {
```

Comment:
I think we should only do it in `trace/debug/info/warn/error` methods, otherwise, it will impact `isTraceEnabled()`-like methods

Reply:
Also, I'm worried that only a few users have MDCs enabled in their pattern (e.g. there is no MDC in Spring Boot by default)

Reply:
Yes this is just an example of what it is possible without a big re-factoring but I think the best option would be to re-think a little logging in general. What I've noticed is that often the same info provided by the logger name is also included in the log message.